### PR TITLE
feat: 🎸 [HCPSDKFIORIUIKIT-2683] [SwiftUI] Slider

### PR DESCRIPTION
**Pull request by JunSong-SH**
This PR addresses changes to support the Slider in the SDK. 

[HCPSDKFIORIUIKIT-2683](https://jira.tools.sap/browse/HCPSDKFIORIUIKIT-2683) [SwiftUI] Slider
[HCPSDKFIORIUIKIT-2684](https://jira.tools.sap/browse/HCPSDKFIORIUIKIT-2684) [SwiftUI] Range Slider

The changes:

- Built new base components: _LowerThumbComponent, _UpperThumbComponent, _ActiveTrackComponent, _InactiveTrackComponent, _LeadingAccessoryComponent and _TrailingAccessoryComponent
- Built new composite components: _RangeSliderControlComponent, _FioriSliderComponent
- Implement the logic to let FioriSlider component support standard slider and range slider both
- Implement the Demo for FioriSlider. 

See below screenshot:

StandardSliderRangeSlider

### CODE REVIEW

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 1]
```diff
@@ -0,0 +1,1039 @@
...
+//  Slider+DataType.swift
+//  FioriSwiftUI
+//
+//  Created by i063052 on 2024/12/13.
+//
+import FioriThemeManager
+import SwiftUI
+
+import Combine
+import FioriThemeManager
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:11]: remove this line

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+import SwiftUI
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:12]: remove this line

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - trailingValueFormat: This optional format is used to format the displayed maximal value of standard slider's range in the trailing accessory view. if provided.
+    ///   - onValueChange: An optional callback function is triggered when the user begins to drag the thumb along the standard slider's track to adjust its value. The first boolean property indicates whether the editing process has begun or ended, with `false` signifying that the editing has concluded. The second double property represents the newly adjusted slider value.
+    ///   - thumb: the Shape for  thumb of the standard Slider. By default, it is Circle.
+    ///   - activeTrack: the Shape for active track of the standard Slider. By default, it is Capsule.
+    ///   - inactiveTrack: the Shape for inactive track of the standard Slider. By default, it is Capsule.
+    ///   - thumbHalfWidth: the half width of the thumb of the standard slider. In the context of a circular representation of the thumb, this value is used as the radius. It should be less than 22. The default value is `14`.
+    ///
+    init(@ViewBuilder title: () -> any View,
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:44]: The init at line 44 and line 126:
What is the major difference btw these two? Is it the extra property of "icon" ?
Can we merge these init into one ?
Same comments apply to other init of "range slider" ,  "single editable range slider", as well.

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+//
+import Combine
+import FioriThemeManager
+import SwiftUI
+
+public extension FioriSlider {
+    /// Create a standard slider
+    ///
+    /// A standard slider consists of a title, a bound value, and a "thumb" (an image that users can drag along a linear "track").
+    /// The track represents a continuum between two extremes: a minimum and a maximum value.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:15]:
```suggestion
    /// The track represents a continuum between two bounded values: a minimum and a maximum value.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 5]
```diff
@@ -0,0 +1,687 @@
...
+    /// By default, the formatted minimum value is displayed at the leading end of the slider, and the formatted maximum value is displayed at the trailing end of the slider.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:16]:
```suggestion
    /// By default, the formatted minimum value is displayed at the leading side of the slider, and the formatted maximum value is displayed at the trailing side of the slider.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 6]
```diff
@@ -0,0 +1,687 @@
...
+    ///
+    /// The title is displayed at the top left of the component, while the bound value is displayed at the top right. As users move the thumb, the slider continuously updates its bound value to reflect the thumb’s position.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:18]:
```suggestion
    /// The title is displayed at the top left of the component, while the bound value is displayed at the top right. As the user moves the thumb, the slider continuously updates its selected value corresponding to the thumb’s position.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 7]
```diff
@@ -0,0 +1,687 @@
...
+    /// You also can customize the slider with customized value label, leading accessory view and trailing accessory view.
+    ///
+    /// - Parameters:
+    ///   - title: The view used to display the title of the slider.
+    ///   - value: The slider which allows the user to select a single value within a specified range (`range`).
+    ///   - range: The range of the slider values. The default is `0...100`
+    ///   - step: The distance between each valid value. The default is `1`.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:25]:
```suggestion
    ///   - step: incremental/decremental value when the thumb changes its position. The default is `1`.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - icon: The icon image for hint text of the slider
+    ///   - description: The hint text of the slider
+    ///   - valueLabel: The customized view to override the default slider value label for the standard slider
+    ///   - showsValueLabel: Indicates whether the value label is to be displayed or not. The default value is `true`
+    ///   - valueFormat: This optional format is used to format the displayed slider value in the value label view. It is also utilized for formatting the accessibility value, if provided.
+    ///   - leadingAccessory: The customized view to override the default leading accessory view which was a text label to display the minimum value of the range
+    ///   - showsLeadingAccessory: Indicates whether the leading accessory view is to be displayed or not. The default value is `true`
+    ///   - leadingValueFormat: This optional format is used to format the displayed minimal value of standard slider's range in the leading accessory view. if provided.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:34]:
```suggestion
    ///   - leadingValueFormat: This optional format is used to format the displayed minimal value of standard slider's range in the leading accessory view, if provided.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 9]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - trailingAccessory: The customized view to override the default trailing accessory view which was a text label to display the maximum value of the range
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:35]:
```suggestion
    ///   - trailingAccessory: The customized view to override the default trailing accessory view which is a text label to display the maximum value of the range
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - showsTrailingAccessory: Indicates whether the trailing accessory view is to be displayed or not. The default value is `true`
+    ///   - trailingValueFormat: This optional format is used to format the displayed maximal value of standard slider's range in the trailing accessory view. if provided.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:37]:
```suggestion
    ///   - trailingValueFormat: This optional format is used to format the displayed maximal value of standard slider's range in the trailing accessory view, if provided.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 11]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - onValueChange: An optional callback function is triggered when the user begins to drag the thumb along the standard slider's track to adjust its value. The first boolean property indicates whether the editing process has begun or ended, with `false` signifying that the editing has concluded. The second double property represents the newly adjusted slider value.
+    ///   - thumb: the Shape for  thumb of the standard Slider. By default, it is Circle.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:39]:
```suggestion
    ///   - thumb: the shape for thumb of the standard Slider. By default, it is circle.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 12]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - activeTrack: the Shape for active track of the standard Slider. By default, it is Capsule.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:40]:
```suggestion
    ///   - activeTrack: the shape for active track of the standard Slider. By default, it is capsule.
```

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift [hunk 13]
```diff
@@ -0,0 +1,687 @@
...
+    ///   - inactiveTrack: the Shape for inactive track of the standard Slider. By default, it is Capsule.
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:41]:
```suggestion
    ///   - inactiveTrack: the shape for inactive track of the standard Slider. By default, it is capsule.
```

**Review by dyongxu — CHANGES_REQUESTED** [review]
Would like to see if possible to consolidate lots of the inits for different slides.

File: Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift (hunk 3 above)
Comment (JunSong-SH) [reply Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:44]: The main difference is that the `AttributedString` type property, such as `title`, which is declared in `BaseComponent` with `@ViewBuilder`, will be used in two constructors: one with the `AttributedString` type and the other with `@ViewBuilder`, following our styled component approach. 

So, there are two constructors both for "standard slider", "range slider" and "single editable range slider". I think the reason is: By using the `AttributedString` type, the app team can implement less code, while with `@ViewBuilder`, the app team can customize it.

Comment (dyongxu) [reply Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift:44]: Would it make sense to merge them into sth like :
    init(title: AttributedString, @ViewBuilder titleView: () -> any View, .......)
 We can also define the priority, if both title and titleView are set by Application,  we will use title , and ignore titleView.

Comment (JunSong-SH) [reply Sources/FioriSwiftUICore/DataTypes/FioriSlider+DataType.swift]: Fixed! Merge those constructor into one for "standard slider", "range slider" and "single editable range slider"

**Comment by JunSong-SH** [comment]
Close this one and use new PR 945 for Slider

**Pull request closed without merging**